language/tools/move-cli/tests/cli_testsuite.rs

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use move_cli::{MOVE_DATA, MOVE_SRC};
+use move_lang::test_utils::*;
+
+use std::{
+    fs::{self, File},
+    io::{self, BufRead},
+    path::Path,
+    process::Command,
+};
+
+/// Basic datatest testing framework for the CLI. It looks for directories under
+/// `tests/testsuite` that contain an `args.txt` file with arguments that the
+/// `move` binary understands (one set of arguments per line). The testing
+/// framework runs the commands, compares the result to the expected output, and
+/// runs `move clean`.
+/// This is designed for testing the CLI itself, but it can also be used to write
+/// tests for user-defined scripts/modules.
+
+pub const CLI_BINARY: &str = "../../../target/debug/move-cli";
+pub const CLI_TESTSUITE_DIR: &str = "tests/testsuite";
+
+const EXP_EXT: &str = "exp";
+
+// if this env var is set, `move clean` will not be run after each test.
+// this is useful if you want to look at the `move_data` produced by
+// a test.
+const NO_MOVE_CLEAN: &str = "NO_MOVE_CLEAN";
+// if either of these env vars is set, the test harness overwrites the
+// old .exp files with the output instead of checking them against the
+// output.
+const UPDATE_BASELINE: &str = "UPDATE_BASELINE";
+const UB: &str = "UB";
+
+fn format_diff(expected: String, actual: String) -> String {
```

Comment:
perhaps we should make this a public function in test utils. Probably fine for now though